fog/overseer/server/src/responses.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Contains responses that are returned by Fog Overseer.
+
+use mc_fog_types::ingest_common::IngestSummary;
+use mc_fog_uri::FogIngestUri;
+use serde::Serialize;
+use std::collections::HashMap;
+
+/// The ingest summaries for the Fog Ingest cluster that Fog Overser is
+/// monitoring. If an ingest summary can't be retrieved for a given node, then
+/// an error message, rather than an ingest summary, is returned for that node.
+#[derive(Serialize)]
+pub struct GetIngestSummariesResponse {
+    pub ingest_summaries: HashMap<FogIngestUri, Result<IngestSummary, String>>,
```

Comment:
Ha, I didn't know `Result` can be serialized to JSON this easily. Neat. Would you mind sharing how the JSON ends up looking like?

Reply:
Yea, I was pleasantly surprised. I guess it makese sense: rocket's Json<T> accepts any T that implements serde::Serialize and Result does. 

Here's a snippet of the output:

```
{
  "ingest_summaries": {
    "insecure-fog-ingest://0.0.0.0:5021/": {
      "Ok": {
        "ingest_controller_mode": "IDLE",
        "next_block_index": 0,
        "pubkey_expiry_window": 10,
        "ingress_pubkey": [
            ...
        ],
        "egress_pubkey": [ 
         ...
        ],
        "kex_rng_version": 0,
        "peers": [
          "insecure-igp://0.0.0.0:5002/",
          "insecure-igp://0.0.0.0:5012/",
          "insecure-igp://0.0.0.0:5022/"
        ],
        "ingest_invocation_id": 0
      }
    },
   ...
}
```